prboom2/src/r_draw.c

```diff
@@ -107,20 +107,26 @@ static const byte   *tempfuzzmap;
 //#define FUZZOFF (SCREENWIDTH)
 #define FUZZOFF 1
 
-static const int fuzzoffset_org[FUZZTABLE] = {
-  FUZZOFF,-FUZZOFF,FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,
-  FUZZOFF,FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,
-  FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,-FUZZOFF,-FUZZOFF,-FUZZOFF,
-  FUZZOFF,-FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,
-  FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,-FUZZOFF,FUZZOFF,
-  FUZZOFF,-FUZZOFF,-FUZZOFF,-FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,
-  FUZZOFF,FUZZOFF,-FUZZOFF,FUZZOFF,FUZZOFF,-FUZZOFF,FUZZOFF
-};
-
-static int fuzzoffset[FUZZTABLE];
-
 static int fuzzpos = 0;
 
+// Fuzz cell size for scaled software fuzz
+static int fuzzcellsize;
+
+// Fuzz colormap table. Contains colormaps for fuzz intensities, where a
+// fuzz intensity is how strong a fuzz cell is, or how much it darkens its
+// contents
+// NOTE: Formula to find a colormap for an intensity is
+// (6 + 21 - (9 - i) * (9 - i) * 5 / 19), where i is the fuzz intensity
+static const byte fuzzcmaps[FUZZTABLE] = {
+  6, 11, 6, 11, 6, 6, 11,
```

Comment:
So the `6` is coming from `tempfuzzmap[6*256+(col)]`, right? But where do the other numbers come from exactly?

Reply:
Even if I did understand what the formula was doing at some point, it was written months ago and I don't really know now.

Reply:
It's important that we don't change the color of the fuzz. No pixel in the software fuzz should have a color that it couldn't have in vanilla. Basically I want to make sure solving the pattern doesn't introduce a separate problem :^)

The sample screenshots appear to have a bit darker shading, but I guess I will have to verify this.

Reply:
Oh yeah the fuzz will be a slightly different shade than vanilla, but only because the colormaps don't offer that amount of granularity. The colormap values were chosen to be the closest fit. If this discrepancy is a problem, I'll see what I can do about getting more precise shades.

Reply:
Now the fuzz's colors are accurate to vanilla